crates/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -587,6 +629,7 @@ impl Context {
     /// Copy the array and generates a write opcode on the new array
     ///
     /// Note: Copying the array is inefficient and is not the way we want to do it in the end.
+    #[allow(clippy::too_many_arguments)]
```

Comment:
This is leftover from when I was still just aiming for functionality, going to refactor this method,

Reply:
I went with a strategy where we pass around the instruction itself rather than the instruction as well as all the internal fields. This leads to having to perform extra match statements, but overall simpler function signatures.